src/y_array.rs

```diff
@@ -196,6 +205,89 @@ impl YArray {
     }
 }
 
+impl YArray {
+    /// Gets a single element from a YArray.
+    fn get_element(&self, index: u32) -> PyResult<PyObject> {
+        match &self.0 {
+            SharedType::Integrated(v) => {
+                if let Some(value) = v.get(index as u32) {
+                    Ok(Python::with_gil(|py| value.into_py(py)))
+                } else {
+                    Err(PyIndexError::new_err(
+                        "Index outside the bounds of an YArray",
+                    ))
+                }
+            }
+            SharedType::Prelim(v) => {
+                if let Some(value) = v.get(index as usize) {
+                    Ok(value.clone())
+                } else {
+                    Err(PyIndexError::new_err(
+                        "Index outside the bounds of an YArray",
+                    ))
+                }
+            }
+        }
+    }
+
+    /// Creates a new YArray from a range of values specified in a PySlice
+    fn get_range(&self, slice: &PySlice) -> PyResult<PyObject> {
```

Comment:
We could create such method at `yrs` level. There may be some questions about applicability of this operation to YText (as naively thinking `ytext[5..10]` would return a string, but this ignores embedded and formatting blocks).

Reply:
The key distinction here is user intention. I can see two use cases here:
1. The developer is quickly extracting a substring, expected to get a `str` in order to feed to a native Python function.
2. The developer is extracting a preliminary substring that they expect to insert into the YCRDT.

I had my mind on the former throughout the PR, but the latter makes a lot of sense, especially because it preserves the rich text markers. So this function could return a YText instance instead of a string. But I would not count the rich text markers as characters. It'd make sense to support some sort of tag autocomplete so that `"<b>slice</b>"[2:] == "<b>ice</b>"`.